pkgs/by-name/re/resnap/package.nix

```
@@ -0,0 +1,48 @@
+{
+  lib,
+  stdenv,
+  writeShellApplication,
+  ffmpeg,
+  feh,
+  imagemagick_light,
+  lz4,
+  fetchFromGitHub,
+}:
+let
+  name = "reSnap";
+  version = "2.5.2";
+
+  src = stdenv.mkDerivation {
+    inherit name version;
+    src = fetchFromGitHub {
+      owner = "cloudsftp";
+      repo = "reSnap";
+      rev = "v${version}";
+      hash = "sha256-thVyf1gTDPLQVtZKoWL7SGiWI++ICWqmF/Ar57I3WP8=";
+    };
```

Since we're just patching and copying a shell script, we only need these phases:

```suggestion
    };
    phases = [
      "unpackPhase"
      "patchPhase"
      "installPhase"
    ];
```

---

Setting phases in packages is strictly forbidden. Also if the phases do not cause harm, just leave them enabled.

---

> Setting phases in packages is strictly forbidden

@SuperSandro2000 According to the [stdenv docs](https://github.com/NixOS/nixpkgs/blob/master/doc/stdenv/stdenv.chapter.md#phases-var-stdenv-phases), it's not forbidden to set this. It's just discouraged.

> It is discouraged to set this variable, as it is easy to miss some important functionality hidden in some of the less obviously needed phases (like `fixupPhase` which patches the shebang of scripts).
Usually, if you just want to add a few phases, it’s more convenient to set one of the variables below (such as `preInstallPhases`).

But perhaps it's that worth it after all.